File: swift/moveZeros.swift
func moveZeroes(inout nums: [Int]) {
    var numOfZeros = 0
    var newArray = [Int]()
    for n in nums {
        if n == 0 {
            numOfZeros += 1
        } else {
            newArray.append(n)
        }
    }
    for _ in 0...numOfZeros {
        newArray.append(0)
    }
    nums = newArray
}

File: ruby/move_zeros.rb
def move_zeroes(nums)
  num_of_zeroes = 0
  new_array = []
  nums.each do |n|
    if n == 0
      num_of_zeroes += 1
    else
      new_array.push n
    end
  end  
  for i in 0..num_of_zeroes
    new_array.push 0
  end
end
